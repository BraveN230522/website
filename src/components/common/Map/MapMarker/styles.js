import React from 'react';
import styled from '@emotion/styled';

const MarkerStyle = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  border: none;

  &:hover,
  &:focus {
    cursor: pointer;
    transform: scale(1.2);
    outline: none;
  }
`;

const SvgMarker = ({ size }) => (
  <svg width={size} height={size} viewBox="0 0 73 73" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g filter="url(#filter0_d)">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M16.8195 14.7947C21.9336 13.0474 28.8542 12 36.4994 12H36.5006C44.1458 12 51.0664 13.0474 56.1805 14.7947C58.7375 15.6683 60.8445 16.709 62.3942 17.9608C63.9243 19.1967 64.9736 20.7371 65 22.5C64.9736 24.2628 63.9243 25.8033 62.3942 27.0392C60.8445 28.291 58.7375 29.3317 56.1805 30.2053C51.0664 31.9526 44.1458 33 36.5006 33H36.4994C28.8542 33 21.9336 31.9526 16.8195 30.2053C14.2625 29.3317 12.1555 28.291 10.6058 27.0392C9.07573 25.8033 8.02638 24.2628 8 22.5C8.02638 20.7371 9.07573 19.1967 10.6058 17.9608C12.1555 16.709 14.2625 15.6683 16.8195 14.7947ZM36.5006 16.2843H36.4994C32.3294 16.2976 13.6393 18.1311 13.4474 22.5C13.6393 26.8689 32.3294 28.7024 36.4994 28.7157H36.5006C40.6706 28.7017 59.3607 26.8689 59.5526 22.5C59.3607 18.1311 40.6706 16.2976 36.5006 16.2843Z"
        fill="black"
      />
      <path
        d="M65 28C64.6473 28.3585 64.2774 28.6966 63.8926 28.999C62.0057 30.4816 59.6788 31.5685 56.9465 32.4766C51.4818 34.2927 44.3617 35.3188 36.4988 35.3188C28.6359 35.3188 21.5206 34.2927 16.0559 32.4766C13.3235 31.5685 10.9907 30.4816 9.10381 28.999C8.71975 28.6972 8.35187 28.3601 8 28.0024V53.1778C8.11908 54.7881 9.15172 56.2064 10.6067 57.3497C11.5569 58.0963 12.719 58.7645 14.0638 59.3666V48.6186C14.0639 47.9545 14.6011 47.4147 15.2671 47.4094L21.691 49.2751C22.2033 49.4283 22.5545 49.8979 22.5556 50.4312V61.8888C25.3342 62.4253 28.3919 62.8094 31.6489 63V51.0358C31.6489 50.3659 32.1898 49.8266 32.8617 49.8266H40.1383C40.8102 49.8266 41.3511 50.3659 41.3511 51.0358V63C44.6089 62.8094 47.6676 62.4255 50.4468 61.8888V50.4312C50.4477 49.8982 50.7984 49.4287 51.3102 49.2751L57.6891 47.4094C58.3731 47.3888 58.9388 47.9362 58.9385 48.6186V59.3654C60.2823 58.7636 61.4437 58.0958 62.3933 57.3497C63.8458 56.2084 64.878 54.7929 65 53.1861V28Z"
        fill="black"
      />
      <path
        d="M57.2389 6.00063C56.5472 6.02554 55.9996 6.58998 56 7.27754V12.7094C56.1353 12.7527 56.282 12.7899 56.4155 12.8341C57.4982 13.1926 58.5199 13.5777 59.4807 14L64.2881 11.6133C65.2373 11.1431 65.2373 9.79772 64.2881 9.32757L57.8602 6.1378C57.668 6.04106 57.4543 5.99389 57.2389 6.00063Z"
        fill="black"
      />
      <path
        d="M14 7.16257V13C14.662 12.757 15.3506 12.5263 16.0649 12.3073C18.1073 11.6811 20.3765 11.1577 22.812 10.7336C23.2035 10.1425 22.9618 9.36991 22.2886 9.06066L15.8603 6.1073C15.001 5.71251 14 6.48477 14 7.16257Z"
        fill="black"
      />
    </g>
    <defs>
      <filter
        id="filter0_d"
        x="0"
        y="0"
        width="73"
        height="73"
        filterUnits="userSpaceOnUse"
        colorInterpolationFilters="sRGB"
      >
        <feFlood floodOpacity="0" result="BackgroundImageFix" />
        <feColorMatrix in="SourceAlpha" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" />
        <feOffset dy="2" />
        <feGaussianBlur stdDeviation="4" />
        <feColorMatrix type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.455256 0" />
        <feBlend mode="normal" in2="BackgroundImageFix" result="effect1_dropShadow" />
        <feBlend mode="normal" in="SourceGraphic" in2="effect1_dropShadow" result="shape" />
      </filter>
    </defs>
  </svg>
);

const HotelMarker = ({ size }) => (
  <svg
    width={size}
    height={size * 1.16}
    viewBox="0 0 68 79"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    // xmlns:xlink="http://www.w3.org/1999/xlink"
    xmlSpace="preserve"
    // xmlns:serif="http://www.serif.com/"
    style={{
      fillRule: 'evenodd',
      clipRule: 'evenodd',
      strokeLinejoin: 'round',
      strokeMiterlimit: 2,
    }}
  >
    <g transform="matrix(1.14286,0,0,1.14286,-21.6243,-33.7143)">
      <path d="M34.5,33L35.286,35.418L37.829,35.418L35.771,36.913L36.557,39.332L34.5,37.837L32.443,39.332L33.229,36.913L31.171,35.418L33.714,35.418L34.5,33Z" />
    </g>
    <g transform="matrix(1.14286,0,0,1.14286,-13.6243,-34.7143)">
      <path d="M34.5,33L35.286,35.418L37.829,35.418L35.771,36.913L36.557,39.332L34.5,37.837L32.443,39.332L33.229,36.913L31.171,35.418L33.714,35.418L34.5,33Z" />
    </g>
    <g transform="matrix(1.14286,0,0,1.14286,-5.62435,-37.7143)">
      <path d="M34.5,33L35.286,35.418L37.829,35.418L35.771,36.913L36.557,39.332L34.5,37.837L32.443,39.332L33.229,36.913L31.171,35.418L33.714,35.418L34.5,33Z" />
    </g>
    <g transform="matrix(1.14286,0,0,1.14286,3.37565,-34.7143)">
      <path d="M34.5,33L35.286,35.418L37.829,35.418L35.771,36.913L36.557,39.332L34.5,37.837L32.443,39.332L33.229,36.913L31.171,35.418L33.714,35.418L34.5,33Z" />
    </g>
    <g transform="matrix(1.14286,0,0,1.14286,10.3757,-33.7143)">
      <path d="M34.5,33L35.286,35.418L37.829,35.418L35.771,36.913L36.557,39.332L34.5,37.837L32.443,39.332L33.229,36.913L31.171,35.418L33.714,35.418L34.5,33Z" />
    </g>
    <g transform="matrix(1,0,0,1,-29,-27)">
      <path d="M37.5,71C41.639,71 45,74.809 45,79.5C45,84.191 41.639,88 37.5,88C33.361,88 30,84.191 30,79.5C30,74.809 33.361,71 37.5,71ZM37.5,75.25C39.57,75.25 41.25,77.154 41.25,79.5C41.25,81.846 39.57,83.75 37.5,83.75C35.43,83.75 33.75,81.846 33.75,79.5C33.75,77.154 35.43,75.25 37.5,75.25Z" />
    </g>
    <g transform="matrix(1,0,0,1,21,-27)">
      <path d="M37.5,71C41.639,71 45,74.809 45,79.5C45,84.191 41.639,88 37.5,88C33.361,88 30,84.191 30,79.5C30,74.809 33.361,71 37.5,71ZM37.5,75.25C39.57,75.25 41.25,77.154 41.25,79.5C41.25,81.846 39.57,83.75 37.5,83.75C35.43,83.75 33.75,81.846 33.75,79.5C33.75,77.154 35.43,75.25 37.5,75.25Z" />
    </g>
    <g transform="matrix(1,0,0,1,-15,-27)">
      <path d="M37,99.5C37,95.913 34.087,93 30.5,93L21.5,93C17.913,93 15,95.913 15,99.5C15,103.087 17.913,106 21.5,106L30.5,106C34.087,106 37,103.087 37,99.5ZM36,99.455C36,98.1 34.9,97 33.545,97L21.455,97C20.1,97 19,98.1 19,99.455C19,100.809 20.1,101.909 21.455,101.909L33.545,101.909C34.9,101.909 36,100.809 36,99.455Z" />
    </g>
    <g transform="matrix(-1,0,0,1,83,-27)">
      <path d="M37,99.5C37,95.913 34.087,93 30.5,93L21.5,93C17.913,93 15,95.913 15,99.5C15,103.087 17.913,106 21.5,106L30.5,106C34.087,106 37,103.087 37,99.5ZM36,99.455C36,98.1 34.9,97 33.545,97L21.455,97C20.1,97 19,98.1 19,99.455C19,100.809 20.1,101.909 21.455,101.909L33.545,101.909C34.9,101.909 36,100.809 36,99.455Z" />
    </g>
    <g transform="matrix(0.75,0,0,0.7,-7,1.9)">
      <rect x="20" y="83" width="4" height="10" />
    </g>
    <g transform="matrix(0.75,0,0,0.7,42,1.9)">
      <rect x="20" y="83" width="4" height="10" />
    </g>
    <g transform="matrix(1,0,0,1,-40,-27)">
      <path d="M62,106L55.504,106C54.122,106 53,104.878 53,103.496L53,43.504C53,42.122 54.122,41 55.504,41L91.496,41C92.878,41 94,42.122 94,43.504L94,103.496C94,104.878 92.878,106 91.496,106L86,106L86,90.175C86,88.975 85.025,88 83.825,88L64.175,88C62.975,88 62,88.975 62,90.175L62,106ZM68,106L68,94.088C68,93.487 68.487,93 69.088,93L78.912,93C79.513,93 80,93.487 80,94.088L80,106L68,106ZM87,75.75C87,75.336 86.664,75 86.25,75L84.75,75C84.336,75 84,75.336 84,75.75L84,82.25C84,82.664 84.336,83 84.75,83L86.25,83C86.664,83 87,82.664 87,82.25L87,75.75ZM79,75.75C79,75.336 78.664,75 78.25,75L76.75,75C76.336,75 76,75.336 76,75.75L76,82.25C76,82.664 76.336,83 76.75,83L78.25,83C78.664,83 79,82.664 79,82.25L79,75.75ZM64,75.75C64,75.336 63.664,75 63.25,75L61.75,75C61.336,75 61,75.336 61,75.75L61,82.25C61,82.664 61.336,83 61.75,83L63.25,83C63.664,83 64,82.664 64,82.25L64,75.75ZM72,76C72,75.448 71.552,75 71,75L69,75C68.448,75 68,75.448 68,76L68,82C68,82.552 68.448,83 69,83L71,83C71.552,83 72,82.552 72,82L72,76ZM87,62.75C87,62.336 86.664,62 86.25,62L84.75,62C84.336,62 84,62.336 84,62.75L84,69.25C84,69.664 84.336,70 84.75,70L86.25,70C86.664,70 87,69.664 87,69.25L87,62.75ZM79,62.75C79,62.336 78.664,62 78.25,62L76.75,62C76.336,62 76,62.336 76,62.75L76,69.25C76,69.664 76.336,70 76.75,70L78.25,70C78.664,70 79,69.664 79,69.25L79,62.75ZM64,62.75C64,62.336 63.664,62 63.25,62L61.75,62C61.336,62 61,62.336 61,62.75L61,69.25C61,69.664 61.336,70 61.75,70L63.25,70C63.664,70 64,69.664 64,69.25L64,62.75ZM72,63C72,62.448 71.552,62 71,62L69,62C68.448,62 68,62.448 68,63L68,69C68,69.552 68.448,70 69,70L71,70C71.552,70 72,69.552 72,69L72,63ZM64,49.75C64,49.336 63.664,49 63.25,49L61.75,49C61.336,49 61,49.336 61,49.75L61,56.25C61,56.664 61.336,57 61.75,57L63.25,57C63.664,57 64,56.664 64,56.25L64,49.75ZM79,49.75C79,49.336 78.664,49 78.25,49L76.75,49C76.336,49 76,49.336 76,49.75L76,56.25C76,56.664 76.336,57 76.75,57L78.25,57C78.664,57 79,56.664 79,56.25L79,49.75ZM87,50C87,49.448 86.552,49 86,49L84,49C83.448,49 83,49.448 83,50L83,56C83,56.552 83.448,57 84,57L86,57C86.552,57 87,56.552 87,56L87,50ZM72,50C72,49.448 71.552,49 71,49L69,49C68.448,49 68,49.448 68,50L68,56C68,56.552 68.448,57 69,57L71,57C71.552,57 72,56.552 72,56L72,50Z" />
    </g>
  </svg>
);

export { MarkerStyle, SvgMarker, HotelMarker };
