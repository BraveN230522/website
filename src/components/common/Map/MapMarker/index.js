import React from 'react';

import MapTooltip from '../MapTooltip';

import { MarkerStyle, SvgMarker, HotelMarker } from './styles';

const MapMarker = ({
  id,
  fromDate,
  dateTime,
  toDate,
  headline,
  venueId,
  venueName,
  venueAddress,
  otherEventsCount,
  hotelId,
  hotelName,
  hotelAddress,
}) => {
  return (
    <>
      <MarkerStyle data-tip data-for={id}>
        {hotelId ? <HotelMarker size={45} /> : <SvgMarker size={45} />}
      </MarkerStyle>
      <MapTooltip
        id={id}
        dateTime={dateTime}
        fromDate={fromDate}
        toDate={toDate}
        headline={headline}
        venueId={venueId || hotelId}
        venueName={venueName || hotelName}
        venueAddress={venueAddress || hotelAddress}
        otherEventsCount={otherEventsCount}
        hideLink={hotelId !== undefined}
      />
    </>
  );
};

export default MapMarker;
