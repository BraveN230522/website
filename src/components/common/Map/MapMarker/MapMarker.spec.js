import React from 'react';
import { render } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import MapMarker from './index';

jest.mock('crypto', () => ({
  randomBytes: num => new Array(num).fill(0),
}));

describe('<MapMarker />', () => {
  it('Renders the MapMarker', () => {
    const { container } = render(
      <MapMarker id="id-123" headline="Great event" venueName="Online" venueAddress="Your web browser" />,
      {
        wrapper: withGlobals,
      }
    );

    expect(container).toMatchSnapshot();
  });
});
