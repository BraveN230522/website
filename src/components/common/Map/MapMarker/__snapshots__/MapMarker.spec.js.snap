// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapMarker /> Renders the MapMarker 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
}

.emotion-0:hover,
.emotion-0:focus {
  cursor: pointer;
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
  outline: none;
}

.emotion-10 {
  position: relative;
  padding: 24px;
  border-radius: 9px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 339px;
  background-color: #ffffff;
}

.emotion-2 {
  opacity: 0.8;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-size: 17px;
  font-weight: bold;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
  color: #2891ff;
}

.emotion-2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-4 {
  opacity: 0.8;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-size: 14px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  color: #000000;
}

.emotion-6 {
  margin: 0;
  padding: 0;
  opacity: 0.8;
  font-family: Inter,sans-serif;
  font-style: normal;
  font-size: 12.3px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  color: #6c6b6b;
}

.emotion-8 {
  position: absolute;
  right: 16px;
  top: 45.5px;
}

.emotion-8:hover {
  cursor: pointer;
}

<div>
  <button
    class="emotion-0 emotion-1"
    currentitem="false"
    data-for="id-123"
    data-tip="true"
  >
    <svg
      fill="none"
      height="20"
      viewBox="0 0 73 73"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        filter="url(#filter0_d)"
      >
        <path
          clip-rule="evenodd"
          d="M16.8195 14.7947C21.9336 13.0474 28.8542 12 36.4994 12H36.5006C44.1458 12 51.0664 13.0474 56.1805 14.7947C58.7375 15.6683 60.8445 16.709 62.3942 17.9608C63.9243 19.1967 64.9736 20.7371 65 22.5C64.9736 24.2628 63.9243 25.8033 62.3942 27.0392C60.8445 28.291 58.7375 29.3317 56.1805 30.2053C51.0664 31.9526 44.1458 33 36.5006 33H36.4994C28.8542 33 21.9336 31.9526 16.8195 30.2053C14.2625 29.3317 12.1555 28.291 10.6058 27.0392C9.07573 25.8033 8.02638 24.2628 8 22.5C8.02638 20.7371 9.07573 19.1967 10.6058 17.9608C12.1555 16.709 14.2625 15.6683 16.8195 14.7947ZM36.5006 16.2843H36.4994C32.3294 16.2976 13.6393 18.1311 13.4474 22.5C13.6393 26.8689 32.3294 28.7024 36.4994 28.7157H36.5006C40.6706 28.7017 59.3607 26.8689 59.5526 22.5C59.3607 18.1311 40.6706 16.2976 36.5006 16.2843Z"
          fill="black"
          fill-rule="evenodd"
        />
        <path
          d="M65 28C64.6473 28.3585 64.2774 28.6966 63.8926 28.999C62.0057 30.4816 59.6788 31.5685 56.9465 32.4766C51.4818 34.2927 44.3617 35.3188 36.4988 35.3188C28.6359 35.3188 21.5206 34.2927 16.0559 32.4766C13.3235 31.5685 10.9907 30.4816 9.10381 28.999C8.71975 28.6972 8.35187 28.3601 8 28.0024V53.1778C8.11908 54.7881 9.15172 56.2064 10.6067 57.3497C11.5569 58.0963 12.719 58.7645 14.0638 59.3666V48.6186C14.0639 47.9545 14.6011 47.4147 15.2671 47.4094L21.691 49.2751C22.2033 49.4283 22.5545 49.8979 22.5556 50.4312V61.8888C25.3342 62.4253 28.3919 62.8094 31.6489 63V51.0358C31.6489 50.3659 32.1898 49.8266 32.8617 49.8266H40.1383C40.8102 49.8266 41.3511 50.3659 41.3511 51.0358V63C44.6089 62.8094 47.6676 62.4255 50.4468 61.8888V50.4312C50.4477 49.8982 50.7984 49.4287 51.3102 49.2751L57.6891 47.4094C58.3731 47.3888 58.9388 47.9362 58.9385 48.6186V59.3654C60.2823 58.7636 61.4437 58.0958 62.3933 57.3497C63.8458 56.2084 64.878 54.7929 65 53.1861V28Z"
          fill="black"
        />
        <path
          d="M57.2389 6.00063C56.5472 6.02554 55.9996 6.58998 56 7.27754V12.7094C56.1353 12.7527 56.282 12.7899 56.4155 12.8341C57.4982 13.1926 58.5199 13.5777 59.4807 14L64.2881 11.6133C65.2373 11.1431 65.2373 9.79772 64.2881 9.32757L57.8602 6.1378C57.668 6.04106 57.4543 5.99389 57.2389 6.00063Z"
          fill="black"
        />
        <path
          d="M14 7.16257V13C14.662 12.757 15.3506 12.5263 16.0649 12.3073C18.1073 11.6811 20.3765 11.1577 22.812 10.7336C23.2035 10.1425 22.9618 9.36991 22.2886 9.06066L15.8603 6.1073C15.001 5.71251 14 6.48477 14 7.16257Z"
          fill="black"
        />
      </g>
      <defs>
        <filter
          color-interpolation-filters="sRGB"
          filterUnits="userSpaceOnUse"
          height="73"
          id="filter0_d"
          width="73"
          x="0"
          y="0"
        >
          <feflood
            flood-opacity="0"
            result="BackgroundImageFix"
          />
          <fecolormatrix
            in="SourceAlpha"
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
          />
          <feoffset
            dy="2"
          />
          <fegaussianblur
            stdDeviation="4"
          />
          <fecolormatrix
            type="matrix"
            values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.455256 0"
          />
          <feblend
            in2="BackgroundImageFix"
            mode="normal"
            result="effect1_dropShadow"
          />
          <feblend
            in="SourceGraphic"
            in2="effect1_dropShadow"
            mode="normal"
            result="shape"
          />
        </filter>
      </defs>
    </svg>
  </button>
  <div
    aria-haspopup="true"
    class="__react_component_tooltip t00000000-0000-4000-8000-000000000000 place-top type-dark allow_hover"
    data-id="tooltip"
    id="id-123"
  >
    <style>
      
  	.t00000000-0000-4000-8000-000000000000 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t00000000-0000-4000-8000-000000000000.place-top {
        margin-top: -10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-top::before {
        border-top: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t00000000-0000-4000-8000-000000000000.place-bottom {
        margin-top: 10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t00000000-0000-4000-8000-000000000000.place-left {
        margin-left: -10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-left::before {
        border-left: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t00000000-0000-4000-8000-000000000000.place-right {
        margin-left: 10px;
    }
    .t00000000-0000-4000-8000-000000000000.place-right::before {
        border-right: 8px solid transparent;
    }
    .t00000000-0000-4000-8000-000000000000.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  
    </style>
    <div
      class="emotion-10 emotion-11"
    >
      <a
        class="emotion-2 emotion-3"
      >
        Great event
      </a>
      <span
        class="emotion-4 emotion-5"
      >
        Online
      </span>
      <p
        class="emotion-6 emotion-7"
      >
        Your web browser
      </p>
      <a
        class="emotion-8 emotion-9"
      >
        <svg
          fill="none"
          height="28"
          viewBox="0 0 16 28"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 2.30739L12.3001 13.999L0.000981601 25.6902L1.79731 27.5L16 13.999L1.79731 0.5L0 2.30739Z"
            fill="#2A98FF"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
